### PR TITLE
docs: add distinct operator documentation

### PR DESCRIPTION
**Pull request by KyleAMathews**
## Summary
- Added comprehensive documentation for the `distinct` query operator to the live queries guide
- Based the documentation on existing test cases in `packages/db/tests/query/distinct.test.ts`

## Changes
- Added new section "Distinct" to the live queries documentation
- Included examples for:
  - Basic single and multi-column distinct usage
  - Computed fields with distinct
  - Functional selects with distinct
  - Combining distinct with orderBy
  - Live updates behavior
  - Edge cases (null values, boolean values)
  - Important notes about requirements and performance

## Test plan
- [x] Documentation is based on existing comprehensive test suite
- [x] All code examples follow the same pattern as other sections in the guide
- [x] Added to table of contents

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: docs/guides/live-queries.md [hunk 1]
```diff
@@ -868,6 +869,187 @@ const engineeringStats = deptStats.get(1)
...
+    .distinct()
+)
+
+// Result will contain at most: [true, false]
+```
+
+### Important Notes
+
+- **Requires Select**: You must use a `select` clause before `distinct`. Attempting to use `distinct` without `select` will throw a `DistinctRequiresSelectError`.
+- **Performance**: The distinct operation is performed after data retrieval and transformation, so it processes all matching rows before deduplication.
```
Comment (samwillis) [inline docs/guides/live-queries.md]: This seems unnecessary to call out
